File: pkg/dev_compiler/test/codegen_expected/language/js_properties_test.js
dart_library.library('language/js_properties_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__js_properties_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const js_properties_test = Object.create(null);
  let VoidTovoid = () => (VoidTovoid = dart.constFn(dart.definiteFunctionType(dart.void, [])))();
  js_properties_test.main = function() {
    expect$.Expect.equals(42, new js_properties_test.__defineGetter__().hello());
    expect$.Expect.equals(42, new js_properties_test.__defineSetter__().hello());
    expect$.Expect.equals(42, new js_properties_test.__lookupGetter__().hello());
    expect$.Expect.equals(42, new js_properties_test.__lookupSetter__().hello());
    expect$.Expect.equals(42, new js_properties_test.constructor().hello());
    expect$.Expect.equals(42, new js_properties_test.hasOwnProperty().hello());
    expect$.Expect.equals(42, new js_properties_test.isPrototypeOf().hello());
    expect$.Expect.equals(42, new js_properties_test.propertyIsEnumerable().hello());
    expect$.Expect.equals(42, new js_properties_test.toLocaleString().hello());
    expect$.Expect.equals(42, new js_properties_test.toString().hello());
    expect$.Expect.equals(42, new js_properties_test.valueOf().hello());
  };
  dart.fn(js_properties_test.main, VoidTovoid());
  js_properties_test.Hello = class Hello extends core.Object {
    hello() {
      return 42;
    }
  };
  dart.setSignature(js_properties_test.Hello, {
    methods: () => ({hello: dart.definiteFunctionType(core.int, [])})
  });
  js_properties_test.__defineGetter__ = class __defineGetter__ extends js_properties_test.Hello {};
  js_properties_test.__defineSetter__ = class __defineSetter__ extends js_properties_test.Hello {};
  js_properties_test.__lookupGetter__ = class __lookupGetter__ extends js_properties_test.Hello {};
  js_properties_test.__lookupSetter__ = class __lookupSetter__ extends js_properties_test.Hello {};
  js_properties_test.constructor = class constructor extends js_properties_test.Hello {};
  js_properties_test.hasOwnProperty = class hasOwnProperty extends js_properties_test.Hello {};
  js_properties_test.isPrototypeOf = class isPrototypeOf extends js_properties_test.Hello {};
  js_properties_test.propertyIsEnumerable = class propertyIsEnumerable extends js_properties_test.Hello {};
  js_properties_test.toLocaleString = class toLocaleString extends js_properties_test.Hello {};
  js_properties_test.toString = class toString extends js_properties_test.Hello {};
  js_properties_test.valueOf = class valueOf extends js_properties_test.Hello {};
  // Exports:
  exports.js_properties_test = js_properties_test;
});

File: tests/language_2/js_properties_test.dart
// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// Test that JavaScript properties on Object can still be classes in
// Dart.

import "package:expect/expect.dart";

void main() {
  Expect.equals(42, new __defineGetter__().hello());
  Expect.equals(42, new __defineSetter__().hello());
  Expect.equals(42, new __lookupGetter__().hello());
  Expect.equals(42, new __lookupSetter__().hello());
  Expect.equals(42, new constructor().hello());
  Expect.equals(42, new hasOwnProperty().hello());
  Expect.equals(42, new isPrototypeOf().hello());
  Expect.equals(42, new propertyIsEnumerable().hello());
  Expect.equals(42, new toLocaleString().hello());
  Expect.equals(42, new toString().hello());
  Expect.equals(42, new valueOf().hello());
}

class Hello {
  int hello() => 42;
}

class __defineGetter__ extends Hello {}

class __defineSetter__ extends Hello {}

class __lookupGetter__ extends Hello {}

class __lookupSetter__ extends Hello {}

class constructor extends Hello {}

class hasOwnProperty extends Hello {}

class isPrototypeOf extends Hello {}

class propertyIsEnumerable extends Hello {}

class toLocaleString extends Hello {}

class toString extends Hello {}

class valueOf extends Hello {}
